src/isar_exr/robotinterface.py

```diff
@@ -231,9 +242,53 @@ def initiate_step(self, step: Step) -> None:
         raise NotImplementedError
 
     def step_status(self) -> StepStatus:
-        # TODO: use currentMissionExecution(robotID)
-        self.logger.error("An invalid interface function was called")
-        raise NotImplementedError
+        try:
+            (mission_status, current_task_id) = (
+                self.api.get_mission_status_and_current_task(settings.ROBOT_EXR_ID)
+            )
+            step_status: StepStatus = ExrStepStatus(mission_status).to_step_status()
+        except NoMissionRunningException:
+            # This is a temporary solution until we have mission status by mission id
+            return MissionStatus.Successful
+        except Exception as e:
+            message: str = "Could not get status of running mission\n"
+            self.logger.error(message)
+            raise RobotMissionStatusException(
+                error_description=message,
+            )
+
+        if current_task_id == "end_mission":
+            return StepStatus.Successful
+
+        if (
+            current_task_id == "start_mission"
+            or current_task_id == None
+            or current_task_id == ""
+        ):
+            return StepStatus.NotStarted
+
+        try:
+            step: List[str] = list(
+                filter(lambda step: current_task_id in step, self.mission_task_ids)
+            )[0]
+            task_index: int = self.mission_task_ids.index(step)
+        except ValueError or IndexError:
+            value_error_message: str = (
+                f"Could not find mission task with ID {current_task_id}\n"
+            )
+            self.logger.error(value_error_message)
+            raise RobotStepStatusException(
+                error_description=value_error_message,
+            )
+
+        if task_index > self.current_mission_task_index:
+            self.current_mission_task_index += 1
+            return StepStatus.Successful
+
+        if step_status == MissionStatus.Cancelled:
```

Comment:
Was this for debugging purposes or do we want it for something?

Reply:
It was for debugging, I can remove it. Although it might be nice to report it elsewhere, perhaps in ISAR.